instrumentation/servlet/servlet-common/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/servlet/ExperimentalSnippetHolder.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.bootstrap.servlet;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ExperimentalSnippetHolder {
+
+  private static final AtomicReference<String> snippet = new AtomicReference<>("");
+
+  private static boolean isSet = false;
+
+  public static void setSnippet(String newValue) {
+    String oldValue = snippet.get();
+    while (!isSet) {
+      isSet = snippet.compareAndSet(oldValue, newValue);
+      oldValue = snippet.get();
+    }
+  }
```

Comment:
with AtomicReference, you don't need `isSet` anymore:

```suggestion
  private static final AtomicReference<String> snippet = new AtomicReference<>();

  public static void setSnippet(String newValue) {
    snippet.compareAndSet(null, newValue);
  }
```